File: PackageManager/src/main/scala/entities/TrackEntity.scala
package entities

import javax.persistence._

/**
  * Created by arthurn on 24.06.17.
  */
@Entity
@Table(name = "track", schema = "public")
class TrackEntity {
  private var id = 0
  private var date = 0
  private var location = ""
  private var status = ""
  private var packageId = 0

  @Id
  @Column(name = "id", nullable = false)
  def getId: Int = id

  def setId(id: Int): Unit = {
    this.id = id
  }

  @Basic
  @Column(name = "date", nullable = false)
  def getDate: Int = date

  def setDate(date: Int): Unit = {
    this.date = date
  }

  @Basic
  @Column(name = "location", nullable = false, length = -1)
  def getLocation: String = location

  def setLocation(location: String): Unit = {
    this.location = location
  }

  @Basic
  @Column(name = "status", nullable = false, length = -1)
  def getStatus: String = status

  def setStatus(status: String): Unit = {
    this.status = status
  }

  override def equals(o: Any): Boolean = {
    if (this.equals(o)) return true
    if (o == null || (getClass ne o.getClass)) return false
    val that = o.asInstanceOf[TrackEntity]
    if (id != that.id) return false
    if (if (date != null) !(date == that.date)
    else that.date != null) return false
    if (if (location != null) !(location == that.location)
    else that.location != null) return false
    if (if (status != null) !(status == that.status)
    else that.status != null) return false
    true
  }

  override def hashCode: Int = {
    var result = id
    result = 31 * result + (if (date != null) date.hashCode
    else 0)
    result = 31 * result + (if (location != null) location.hashCode
    else 0)
    result = 31 * result + (if (status != null) status.hashCode
    else 0)
    result
  }

  def getPackageId: Int = packageId

  def setPackageId(packageId: Int): Unit = {
    this.packageId = packageId
  }
}

File: app/src/Entity/TrackEntity.java
package Entity;

import javax.persistence.*;
import java.sql.Timestamp;

/**
 * Created by arthurn on 24.06.17.
 */
@Entity
@Table(name = "track", schema = "public", catalog = "dav5bc69lqbe93")
public class TrackEntity {
    private int id;
    private Timestamp date;
    private String location;
    private String status;
    private PackageEntity packageId;

    @Id
    @Column(name = "id", nullable = false)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "date", nullable = false)
    public Timestamp getDate() {
        return date;
    }

    public void setDate(Timestamp date) {
        this.date = date;
    }

    @Basic
    @Column(name = "location", nullable = false, length = -1)
    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    @Basic
    @Column(name = "status", nullable = false, length = -1)
    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TrackEntity that = (TrackEntity) o;

        if (id != that.id) return false;
        if (date != null ? !date.equals(that.date) : that.date != null) return false;
        if (location != null ? !location.equals(that.location) : that.location != null) return false;
        if (status != null ? !status.equals(that.status) : that.status != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (date != null ? date.hashCode() : 0);
        result = 31 * result + (location != null ? location.hashCode() : 0);
        result = 31 * result + (status != null ? status.hashCode() : 0);
        return result;
    }

    @OneToOne
    @JoinColumn(name = "package_id", referencedColumnName = "id", nullable = false)
    public PackageEntity getPackageId() {
        return packageId;
    }

    public void setPackageId(PackageEntity packageId) {
        this.packageId = packageId;
    }
}
